lib/extract_metadata.py

```diff
@@ -404,7 +404,7 @@ def getMapData(usync, filename, idx, archiveh, springname):
 	res['Version']=version
 	return res
 
-def saveImage(image, size, imagedir):
+def saveImage(image, size, imagedir, createThumbnail=None):
```

Comment:
It could have been done like that, but:
- I already used a script to generate thumbnails for all existing maps
[mapThumbnails.zip](https://github.com/spring/upq/files/9756554/mapThumbnails.zip)

- the metadata extraction which generates the thumbnail together with the other images runs every time a new map is uploaded

there's no situation where that web service would do anything. It's also more efficient to just have the thumbnails there than to have the image.py check if the files exist.

atm there's about 2200 maps but all those thumbnails only take about 25 MB disk space so there's no point in only generating them on-demand.

Reply:
the duplicate code was not included in the PR, just an external script that dealt with the initial situation and won't be needed again any time soon. If we decided to change the thumbnail size  i'd just do a PR with the updated constant for the thumbnail size and rerun the updated script. I don't think we'll need more thumbnail sizes for maps any time soon (possibly many years).

Anyway, if i were to update the PR with that image.py, what exactly would be the parameters?
in your example:
image.py?name=DSD.jpg&width=300&height=300

seems safer to assume it's just the name for a file in the same directory as the map image thumbnails. I'd put the size in a constant within the script and not specify and not check whether the existing image matches the specified restrictions or not (no resizing of already existing thumbnails).

Reply:
This PR's changes have been deployed almost two months ago and have been working fine, I think this PR should be merged as is.

I've been busy with unrelated stuff and am not into refactoring the code again. That dedicated script for resizing images could be specified/implemented in another issue/PR.